### PR TITLE
5139 minmax bad printing

### PR DESCRIPTION
**Pull request by jul-stas**
This is a partial solution to #5139 (only for two types) because of the above and because collections are much harder to do. They are coming in a separate PR.
What this PR does is simple: adds `system.min()` and `system.max()` fns for `time_type` and `inet_address`, so 'overload resolution' does not use `blob->blob` overload, which caused results to be printed as blobs.

### CODE REVIEW

**Comment by denesb** [comment]
On Thu, 2019-11-21 at 07:36 -0800, nyh wrote:
> @nyh commented on this pull request.
> 
> 
> Looks good to me, but:
> 
> 
> It needs the Seastar patch first, and it isn't in yet.
> If the Seastar patch changes the sort order, this one needs to change
> as well.
> 
> Note that 2 is slightly worrying: I means that if Scylla needs a
> specific ordering (perhaps to be 100% compatible with what Cassandra
> does in this case?) we need Seastar's ordering to be the same. Could
> we somehow define Scylla's require ordering as a seperate comparator
> object, instead of operator<?

How would that guarantee compatible ordering? If it relies on the
seastar operator< it could change without notice just the same. We rely
on other libs (most prominently libstdc++) for other orderings as well.
The only guarantee is open coding our own comparators, but even better
is writing tests that verify the expected ordering.

**Comment by avikivity** [comment]
Most types have natual orders and we don't expect a library to just change it.

Where there is no natural order, or if Cassandra's ordering is not the same as the natural order, then we must open-code our comparator to insulate ourselves from the problem.

**Comment by jul-stas** [comment]
I was considering to define this comparison in scylla, but:
- having `operator<` for `inet_address`es solves this problem right away, with minimal code, without messing with implementation of aggregates. (simplicity, safety)
- Due to ADL, the operator should be in `namespace seastar` and I don't know any trick to work around this (anyone?).

Anyway, supplying comparator object to aggregare fns wouldn't be a big deal. The question is if it's better..?

**Comment by denesb** [comment]
I think that for those types that don't have a natural ordering and yet we want to guarantee a certain one (compatible with C*), we can just add specific unit tests. The tests will make sure the ordering will not change unnoticed.

**Comment by jul-stas** [comment]
@denesb  @avikivity @nyh Let's make a conclusion or this will hang forever.
Custom comparator in scylla and no `inet_address::operator<`?

**Comment by nyh** [comment]
@jul-stas I think it is easiest and safest to include this operator< or comparator in Scylla, not in Seastar. The problem isn't just making sure it doesn't change in Seastar; As you said, this would be easy to ensure by using unit tests. The bigger problem is getting the Seastar committers to agree to commit a suspicious-looking operator< that has no real justification besides "this is what Cassandra does". As you can see, your patch to do this was not committed yet into Seastar by any of its committers.

By the way, following up on your question: using "namespace seastar" in Scylla would be "cheating" but an acceptable form of cheating, I think. We already have a bunch of these in Scylla, and even "namespace std"! An example of the latter is how we define a has function in bytes.hh.

**Comment by jul-stas** [comment]
@denesb @nyh Updated - please have a look. Logic of ordering (or lack thereof) is still the same, but now defined in scylla.

File: cql3/functions/aggregate_fcts.hh [hunk 1]
```diff
@@ -247,6 +247,25 @@ struct aggregate_type_for<timeuuid_native_type> {
...
 
+template<>
+struct aggregate_type_for<time_native_type> {
+    using type = time_native_type::primary_type;
+};
+
+template <typename Type>
+const Type& max_wrapper(const Type& t1, const Type& t2) {
+    using std::max;
+    return max(t1, t2);
```
Comment (nyh) [inline cql3/functions/aggregate_fcts.hh]: nitpick: why do with two lines what you can do with one, i.e., call "std::max()" directly?

Comment (jul-stas) [reply cql3/functions/aggregate_fcts.hh]: Thanks! I use this trick sometimes to allow custom implementations, e.g. `max(const WeirdType& wt1, const WeirdType& wt2)` in the namespace where `WeirdType` belongs. Then ADL will prefer the custom over `std::max()`.
But you're right, at this moment it does the same thing, it's only for future-proofness.

**Pull request merged**